### PR TITLE
feat: from doughnut to pie

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

relates to #11433 

As a first step 

* slims the donut
* makes the total underneath it larger
* starts splitting out the doughnut chart from the linegraph
* magnifies an arc on hover

### without legend

Screenshot 2022-09-15 at 16 48 46

### with legend

Screenshot 2022-09-15 at 16 48 56

### CODE REVIEW

**Comment by benjackwhite** [comment]
I think this _looks_ better but I find the small hover area somewhat frustrating... I almost wonder if this isn't better as an option rather than the default? I dunno maybe thats just me

**Comment by pauldambra** [comment]
|chunky|slim|
|-|-|
|Screenshot 2022-09-16 at 14 37 49|Screenshot 2022-09-16 at 14 38 40|

Yep, I went almost all the other way from chunky to slim :)

chartjs expresses it as a percentage of the circle cut out so we can pick a different cutout amount

Pie charts are a bad way of displaying data (IMHO) so the next steps of making the breakdown easier to read are the important ones

**Review by mariusandra — CHANGES_REQUESTED** [review]
**tldr**: style 📈 , usability 📉📉 ... can someone convince me otherwise?

*(I realise I could have voiced this in the design phase, but here we are...)*

I'm having doubts when looking at these screenshots:

|chunky|slim|
|-|-|
|Screenshot 2022-09-16 at 14 37 49|Screenshot 2022-09-16 at 14 38 40|

I find it much easier to conceptualise the fraction when looking at a chunkier donut... versus when looking at the slim ring. I feel like the information loses fidelity the thinner it gets. With a chunkier donut, I find it much easier to conceptualise the fractions. 

Take the sectors next to the red lines:

Screenshot 2022-09-19 at 13 24 17

I can't tell which one is the smallest by glancing at them. I have to stare and measure (I have to [think](https://www.amazon.com/Dont-Make-Think-Revisited-Usability/dp/0321965515)). 

With a chunkier donut, there's much. more for my eyes to hold on to, and the size difference is easier to grasp.

Perhaps they're just *too* thin now, and there's a happy middle ground?

**Comment by lottiecoxon** [comment]
I agree with you guys on this - the usability has massively done down hill in the aim of looking good- need to go back to the drawing board and see what we can do to fix this. Would people be open to having a filled pie chart? this would help with the visibility as well as usability ?

**Comment by pauldambra** [comment]
0 to 100 cutout in 10% steps


cc @lottiecoxon 

0-cutout
10-cutout
20-cutout
30-cutout
40-cutout
50-cutout
60-cutout
70-cutout
80-cutout
90-cutout
100-cutout

**Comment by lottiecoxon** [comment]
0%, 60% and 70% I am happy with 

I was thinking about how to fix our hover issue with the super skinny charts and I was wondering would a layering method help or just be confusing and difficult to implement. (Feel free to pushback- these are just experimental ideas for now)

### Having an invisible selection area 
Idea: when you would hover over the invisible boundary for one of these sections it would react on the visible colour section. 

Untitled_Artwork 116

### Having a slightly faded area filled and a thinner section full 
This would basically create more space for hovering/reacting but with less of the intense colour?

Untitled_Artwork 219

**Comment by pauldambra** [comment]
cc @mariusandra @lottiecoxon as a pie as agreed in https://posthog.slack.com/archives/C0368RPHLQH/p1663942196109679

**Comment by lottiecoxon** [comment]
I love the pie ❤️

**Review by mariusandra — APPROVED** [review]
Cakes at 100% look slightly broken when hovering:



otherwise, LGTM

File: frontend/src/scenes/insights/views/LineGraph/PieChart.tsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+import { insightLogic } from 'scenes/insights/insightLogic'
+import { SeriesDatum } from 'scenes/insights/InsightTooltip/insightTooltipUtils'
+import { SeriesLetter } from 'lib/components/SeriesGlyph'
+
+if (registerables) {
+    // required for storybook to work, not found in esbuild
+    Chart.register(...registerables)
+}
+Chart.register(CrosshairPlugin)
+Chart.defaults.animation['duration'] = 0
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply frontend/src/scenes/insights/views/LineGraph/PieChart.tsx]: we can at least move this to an included file... so at least we don't accidentally change it on one place without realising it's "required" by two

File: frontend/src/scenes/insights/views/LineGraph/PieChart.tsx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                                        {...tooltipConfig}
+                                    />,
+                                    tooltipEl
+                                )
+                            }
+
+                            setTooltipPosition(chart, tooltipEl)
+                        },
+                    },
+                } as _DeepPartialObject<PluginOptionsByType<keyof ChartTypeRegistry>> & ChartPluginsOptions,
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply frontend/src/scenes/insights/views/LineGraph/PieChart.tsx]: And turns out at some point in the changes has become unnecessary 🤷 🤣

**Comment by pauldambra** [comment]
without hover offset when 100%

pie2
